main_test.go

```diff
@@ -0,0 +1,377 @@
+		progs, err := loadPatches(command.fset, opts, os.Stdin)
+		patchRunner := newPatchRunner(command.fset, progs)
+		_, actualComments, _ := patchRunner.Apply(filename, fast)
+		expectedComments := []string{" This patch replaces instances of fmt.Sprintf()",
+			" with fmt.Errorf()", " Patch files can be applied to mutiple files"}
+		require.NoError(t, err)
+		assert.Equal(t, expectedComments, actualComments)
+	})
+	t.Run("Verify Apply functionality when patch doesn't apply", func(t *testing.T) {
+		_, opts := newArgParser()
+		filename := "testdata/test_files/lint_example/time.go"
+		content, _ := ioutil.ReadFile(filename)
+		fast, _ := parser.ParseFile(command.fset, filename, content /* src */, parser.AllErrors|parser.ParseComments)
+		progs, _ := loadPatches(command.fset, opts, os.Stdin)
+		patchRunner := newPatchRunner(command.fset, progs)
+		_, actualComments, ok := patchRunner.Apply(filename, fast)
+		expectedComments := []string(nil)
+		assert.Equal(t, expectedComments, actualComments)
+		assert.Equal(t, false, ok)
+	})
+
+}
+
+func TestPreviewer(t *testing.T) {
+	t.Parallel()
+	t.Run("run functionality with -l", func(t *testing.T) {
+		var tests = []struct {
+			name           string
+			give           []string
+			want           string
+			targetFilename string
+		}{
+			{
+				name:           "Verify preview's functionality",
+				give:           []string{"single line", "different line"},
+				targetFilename: "testdata/test_files/lint_example/time.go",
+				want: "\n" + "--- testdata/test_files/lint_example/time.go\n" +
+					"+++ testdata/test_files/lint_example/time.go\n@@ -1,1 " +
+					"+0,0 @@\n-single line\n@@ -0,0 +1,1 @@\n+different line\n",
+			},
+			{
+				name:           "Verify preview's functionality - empty file",
+				give:           []string{"", ""},
+				targetFilename: "testdata/test_files/lint_example/time.go",
+				want: "\n" + "--- testdata/test_files/lint_example/time.go\n" +
+					"+++ testdata/test_files/lint_example/time.go\n",
+			},
+		}
+		for _, tt := range tests {
+			t.Run(tt.name, func(t *testing.T) {
+				buf := bytes.Buffer{}
+				comment := []string{"example comment"}
+				err := preview(tt.targetFilename, []byte(tt.give[0]), []byte(tt.give[1]), comment, &buf)
+				outputString := buf.String()
+				tt.want = strings.Join(comment, "\n") + tt.want
+				require.NoError(t, err)
+				assert.Equal(t, tt.want, outputString)
+			})
+		}
+
+	})
+
+	t.Run("Verify preview's functionality - nil for strings", func(t *testing.T) {
+		comment := []string{"example comment"}
+		filename := "testdata/test_files/lint_example/time.go"
+
+		buf := bytes.Buffer{}
+		err := preview(filename, nil, nil, comment, &buf)
+		outputString := buf.String()
+		expectedString := strings.Join(comment, "/n") + "\n" + "--- testdata/test_files/lint_example/time.go\n+++ " +
+			"testdata/test_files/lint_example/time.go\n"
+		require.NoError(t, err)
+		assert.Equal(t, expectedString, outputString)
+	})
+}
+
+func TestModifier(t *testing.T) {
```

Comment:
These tests just call modify, which is just ioutil.WriteFile.
I would delete this test function—esp once we inline modify.